shared/test/loop_mocha_utils.js

```diff
@@ -80,14 +90,18 @@ var LoopMochaUtils = (function(global, _) {
   SyncThenable.all = function(promises) {
     return new SyncThenable(function(resolve) {
       var results = [];
+      let resultCount = 0;
 
-      promises.forEach(function(promise) {
+      promises.forEach(function(promise, i) {
         promise.then(function(result) {
-          results.push(result);
+          results[i] = result;
```

Comment:
It might be worth a comment that the index from the iterable is used intentionally to maintain the semantic of Promise.all that guarantees the results are in the order passed in rather than the order resolved, just so that if someone needs to change this code in the future, they don't inadvertently break that.